shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -293,6 +312,33 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
   return YES;
 }
 
+- (UniqueAotDataPtr)loadAotData:(std::string)assetsDir {
+  if (!FlutterEngineRunsAOTCompiledDartCode()) {
+    return nullptr;
+  }
+
+  std::filesystem::path assetsFsDir(assetsDir);
+  std::filesystem::path elfFile("app_elf_snapshot.so");
+  auto fullElfPath = assetsFsDir / elfFile;
+
+  if (!std::filesystem::exists(fullElfPath)) {
+    return nullptr;
+  }
+
+  FlutterEngineAOTDataSource source = {};
+  source.type = kFlutterEngineAOTDataSourceTypeElfPath;
+  source.elf_path = fullElfPath.c_str();
+
+  FlutterEngineAOTData data = nullptr;
+  auto result = FlutterEngineCreateAOTData(&source, &data);
```

Comment:
Doesn't FlutterEngineCreateAOTData require a non-null `data_out` param?

Reply:
It does, but this is a non-null pointer to a pointer variable that will be populated on a successful result. On success, the ownership of that variable is given to a unique_ptr with a custom deleter.

Reply:
Ah yep -- lgtm! This seems fine then but someone needs to clean up the struct.